impeller/display_list/dl_dispatcher.cc

```diff
@@ -1368,22 +1368,52 @@ Canvas& ExperimentalDlDispatcher::GetCanvas() {
 //// Text Frame Dispatcher
 
 TextFrameDispatcher::TextFrameDispatcher(const ContentContext& renderer,
-                                         const Matrix& initial_matrix)
-    : renderer_(renderer), matrix_(initial_matrix) {}
+                                         const Matrix& initial_matrix,
+                                         const Rect cull_rect)
+    : renderer_(renderer), matrix_(initial_matrix) {
+  cull_rect_state_.push_back(cull_rect);
+}
 
-void TextFrameDispatcher::save() {
+TextFrameDispatcher::~TextFrameDispatcher() {
+  FML_DCHECK(cull_rect_state_.size() == 1);
+}
+
+void TextFrameDispatcher::Save(bool push_cull_rect) {
   stack_.emplace_back(matrix_);
+  if (push_cull_rect) {
+    cull_rect_state_.push_back(cull_rect_state_.back());
```

Comment:
An easier way to do this would be to have Save() always push it, and then SL replaces the value on ToS (`.back() = foo`) when it updates it?

Reply:
Done. Testing now